### PR TITLE
[ci] Add Release Drafter workflow

### LINKED ISSUE #6693: [ci] Release drafter stopped working
New merged PRs in master are not listed in the new release draft.

I guess we should transform https://github.com/microsoft/LightGBM/blob/master/.github/release-drafter.yml file into GitHub workflow as it's suggested here: https://github.com/marketplace/actions/release-drafter#usage. Probably Release Drafter **GitHub App** was deprecated and disabled.

### PR DESCRIPTION
**Pull request by StrikerRUS**
Fixed #6693.

Just tested this workflow against non-master branch and it seems to work:

image

### CODE REVIEW

File: .github/workflows/no_response.yml
Comment (StrikerRUS) [inline .github/workflows/no_response.yml:1]: Just for the consistency with other workflow names.

**Comment by StrikerRUS** [comment]
@shiyu1994 Could you please remove from this repo Release Drafter **GitHub App** which is not working ?

File: .github/workflows/release_drafter.yml
Comment (jameslamb) [inline .github/workflows/release_drafter.yml:16]:
```suggestion
    runs-on: ubuntu-latest
```

Is it necessary to pin to a specific, older Linux version? If not, could we just use `ubuntu-latest` so this doesn't need to be updated whenever GitHub drops Ubuntu 22.04 runners?

**Review by jameslamb — APPROVED** [review]
Thank you for fixing this!! I left one small suggestion, but happy to defer to whatever you want to do with it... marking this "approve" so you can merge whenever you want without needing a re-review.

File: .github/workflows/release_drafter.yml
Comment (StrikerRUS) [reply .github/workflows/release_drafter.yml:16]: It's a copy-paste artifact from `workflows/no-response.yml`.

I guess it was there since #5288, so now we can safely use `latest`

**Comment by shiyu1994** [comment]
> @shiyu1994 Could you please remove from this repo Release Drafter **GitHub App** which is not working ?

Done.

**Comment by StrikerRUS** [comment]
@shiyu1994 
> Done.

Thanks a lot!

**Comment by StrikerRUS** [comment]
@jameslamb Could you please help with re-running Azure Pipelines? Or I can close-reopen PR later today.

**Comment by jameslamb** [comment]
> @jameslamb Could you please help with re-running Azure Pipelines? Or I can close-reopen PR later today.

I see it's passing now, so someone else must have restarted it for you 😊 

@shiyu1994 sent a message yesterday in the maintainer Slack... he's still trying to get you access to Azure DevOps. Let me know if you want a new invite to that Slack, in case you lost access.

Until we get you access, please do keep `@`-ing me, I'm happy to re-run these for you.

**Pull request merged**
